e2e/tests/migration_steps_test.go

```diff
@@ -90,45 +219,381 @@ func (s *migrationSuite) TheShannonServiceConfigIsUpdatedIfApplicable(actorType
 }
 
 func (s *migrationSuite) TheAuthorityExecutes(commandStr string) {
-	s.Skip("TODO_UPNEXT(@bryanchriswhite, #1034): Implement.")
+	// DEV_NOTE: If the command doesn't start with "poktrolld"
+	commandStringParts := strings.Split(commandStr, " ")
+	if len(commandStringParts) < 0 && commandStringParts[0] != "poktrolld" {
+		s.Fatalf("ERROR: expected a poktrolld command but got %q", commandStr)
+	}
+
+	// Remove the "poktrolld" part of the command string because
+	// s.pocketd.RunCommand() provides this part of the final command string.
+	commandStringParts = commandStringParts[1:]
+
+	var (
+		results *commandResult
+		err     error
+	)
+	switch {
+	case strings.Contains(commandStr, "import-morse-accounts"):
+		rpcURL, err := url.Parse(defaultRPCURL)
+		require.NoError(s, err)
+
+		grpcAddrFlagString := fmt.Sprintf(
+			"--grpc-addr=%s:%d",
+			rpcURL.Hostname(),
+			defaultGRPCPort,
+		)
+		grpcAddrFlagParts := strings.Split(grpcAddrFlagString, "=")
+		commandStringParts = append(commandStringParts,
+			"--from", "pnf",
+			keyRingFlag,
+			chainIdFlag,
+		)
+		commandStringParts = append(commandStringParts, grpcAddrFlagParts...)
+		results, err = s.pocketd.RunCommandOnHost("", commandStringParts...)
+	default:
+		results, err = s.pocketd.RunCommand(commandStringParts...)
+	}
+
+	require.NoError(s, err)
+	if strings.Contains(results.Stdout, cmdUsagePattern) {
```

Comment:
```suggestion
	require.NoError(s, err)

	// Check if the string provided executed a real command by searching for an expected pattern in the result.
	// It will return a zero exit code on both success and failure.
	if strings.Contains(results.Stdout, cmdUsagePattern) {
```